stable/yugaware/templates/statefulset.yaml

```diff
@@ -538,6 +538,23 @@ spec:
           ports:
             - containerPort: 9000
               name: yugaware
+          {{- if .Values.yugaware.pod.probes.enabled }}
+          startupProbe:
+            failureThreshold: {{ div .Values.yugaware.pod.probes.startupTimeSec.max (div .Values.yugaware.pod.probes.startupTimeSec.min 2) }}
+            initialDelaySeconds: {{ .Values.yugaware.pod.probes.startupTimeSec.min }}
+            periodSeconds: {{ div .Values.yugaware.pod.probes.startupTimeSec.min 2 }}
+            timeoutSeconds: {{ div .Values.yugaware.pod.probes.startupTimeSec.min 4 }}
+            httpGet:
```

Comment:
Can you describe the logic behind the derived values? 
Do we need to derive these values? Can we not have them defined separately in values.yaml?

Reply:
I converted an awkward k8s setting to something meaningful for startup probe parameters.
Startup probe suites to tolerate long application startups before enabling liveness and readiness probes.
From that perspective, 2 numbers are interesting: minimum time app startup typically takes and maximum time, as we want an app to be available asap, but wait long enough to swallow long startups. _k8s doesn't add pod to service up until it passes startup probe in absense of liveness_
Minimum time is put to `initialDelaySeconds`. First probe happens after this delay.
In case of YBA, minimum startup time divided by two makes a good fit for `periodSeconds`. E.g. 30 sec initial delay ~ 15 sec period, 20 sec delay ~ 10 sec period, etc
In order to wait as long as is in `max`, we need to could # of periods in `max`. That's what `{{ div .Values.yugaware.pod.probes.startupTimeSec.max (div .Values.yugaware.pod.probes.startupTimeSec.min 2) }}` does.

Do you want me to dump this to the chart itself ?

Reply:
Thanks for the explaination, i think adding this to commit message would be good enough.